lib/src/trie/trie_node.rs

```diff
@@ -157,10 +165,138 @@ pub fn encode_to_vec(
     Ok(result)
 }
 
+/// Calculates the Merkle value of the given node.
+///
+/// `is_root_node` must be `true` if the encoded node is the root node of the trie.
+///
+/// This is similar to [`encode`], except that the encoding is then optionally hashed.
+///
+/// Hashing is performed if the encoded value is 32 bytes or more, or if `is_root_node` is `true`.
+/// This is the reason why `is_root_node` must be provided.
+pub fn calculate_merkle_value(
+    decoded: Decoded<
+        '_,
+        impl ExactSizeIterator<Item = nibble::Nibble> + Clone,
+        impl AsRef<[u8]> + Clone,
+    >,
+    is_root_node: bool,
+) -> Result<MerkleValueOutput, EncodeError> {
+    // The partial key can only be empty if the node is the root.
+    debug_assert!(is_root_node || decoded.partial_key.len() != 0);
```

Comment:
This assertion is triggered, which seems to indicate a pretty nasty issue in `trie_structure.rs`

Reply:
Ok nevermind, I'm an idiot, this is in fact completely valid.